File: src/main/java/lambda/function/AtomFunction.java
package lambda.function;

import lambda.Function;
import lambda.core.datamodels.AtomNode;

interface AtomFunction extends Function {

    static AtomFunction newInstance() {
        return params -> {
            final var first = params.pop();
            return new AtomNode(first instanceof AtomNode);
        };
    }
}

File: src/main/java/lambda/function/AtomFunction.kt
package lambda.function

import lambda.Function
import lambda.core.datamodels.AtomNode
import lambda.core.datamodels.Stack
import lambda.core.datamodels.Node

internal class AtomFunction : Function {

    override fun evaluate(
        params: Stack<Node>
    ): Node {
        val first = params.pop()
        return AtomNode(first is AtomNode)
    }
}